self_hosted/main.jou

```diff
@@ -235,6 +235,9 @@ class Compiler:
                     if strcmp(self->files[isrc].ast.path, imp->resolved_path) == 0:
                         src = &self->files[isrc]
                         break
+                if (src->ast.path == dest->ast.path):
```

Comment:
I think the same also applies in Jou. Also, you don't need parentheses in Jou if statements. (I think there's an issue about generating a warning about unnecessary parentheses.)

```suggestion
                if src == dest:
```

Reply:
LOL, I write too much C